pkg/codegen/dotnet/utilities.go

```diff
@@ -118,3 +119,23 @@ func makeSafeEnumName(name, typeName string) (string, error) {
 
 	return safeName, nil
 }
+
+// Provides code for a method which will be placed in the program preamble if deemed
+// necessary. Because many Terraform functions are complex, it is much prettier to
+// encapsulate them as their own function in the preamble.
+func getHelperMethodIfNeeded(functionName string) (string, bool) {
+	var methodBody string
+
+	switch functionName {
+	case "sha1":
+		methodBody =
+			`private static string sha1(string input) {
+		return BitConverter.ToString(
+			new SHA1Managed().ComputeHash(Encoding.UTF8.GetBytes(%v))
```

Comment:
The latest .NET version marks `SHA1Managed` obsolete and says to just use `SHA1.Create`:

https://github.com/dotnet/runtime/blob/3633a7d0930abaca701385c1059b80ca157e98c6/src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/SHA1Managed.cs#L9

https://github.com/dotnet/runtime/blob/35cf750ea50f13f215e9eb6d3ecb6dea48b081d1/src/libraries/Common/src/System/Obsoletions.cs#L75

Let's just go ahead and do that here.

```suggestion
			SHA1.Create().ComputeHash(Encoding.UTF8.GetBytes(%v))
```